internal/epoll/poller.go

```diff
@@ -128,7 +129,12 @@ func (p *Poller) Wait(events []unix.EpollEvent) (int, error) {
 	}
 
 	for {
-		n, err := unix.EpollWait(p.epollFd, events, -1)
+		msec := int(-1)
+		if !deadline.IsZero() {
+			msec = int(time.Until(deadline).Milliseconds())
```

Comment:
Bug in my own code: this doesn't handle deadlines in the past or so far in the future that msec is overflowed.

I'll fix that up in a follow up.